### PR TITLE
Pushes builder images to official Dockerhub location

### LINKED ISSUE #18: Full Builder should get published to official DockerHub location
**Context**
Once we've implemented https://github.com/paketo-buildpacks/builder/issues/27 and https://github.com/paketo-buildpacks/builder/issues/28, **and** we've verified that the changes look good, we should swap over from the temporary GCR path to the **official** Dockerhub location for the Full Builder.

Builders should get published to the following location:
`paketobuildpacks/builder:full` (for the latest version)
`paketobuildpacks/builder:<VERSION>-full` (for specific versions)

**Note** At this point, we can get rid of the Concourse jobs for the Full builder.

### PR DESCRIPTION
**Pull request by floragj**
Resolves #18 
Note that we'll need to coordinate with @martyspiewak @mdelillo and @dumez-k before merging this in.

### CODE REVIEW

File: .github/workflows/push-image.yml [hunk 1]
```diff
@@ -29,17 +29,17 @@ jobs:
...
       run: |
-        echo "${GCR_PUSH_BOT_JSON_KEY}" | docker login --username _json_key --password-stdin gcr.io
-        docker tag builder-tmp "gcr.io/paketo-buildpacks/builder-tmp:full"
-        docker tag builder-tmp "gcr.io/paketo-buildpacks/builder-tmp:latest"
-        docker tag builder-tmp "gcr.io/paketo-buildpacks/builder-tmp:${{ steps.event.outputs.tag }}-full"
-
-        docker push "gcr.io/paketo-buildpacks/builder-tmp:full"
-        docker push "gcr.io/paketo-buildpacks/builder-tmp:latest"
-        docker push "gcr.io/paketo-buildpacks/builder-tmp:${{ steps.event.outputs.tag }}-full"
+        echo "${PAKETO_BUILDPACKS_DOCKERHUB_PASSWORD}" | docker login --username paketobuildpacks --password-stdin
```
Comment (ekcasey) [inline .github/workflows/push-image.yml:38]: I feel like its generally good practice to store the username in a secret as well... not that it is particularly secret in this case. Also, have we thought about converting `paketobuildpacks` from a community user to a community organization so we could then log in as a  member. https://docs.docker.com/docker-hub/orgs/

**Review by ekcasey — APPROVED** [review]

File: .github/workflows/push-image.yml (hunk 1 above)
Comment (martyspiewak) [reply .github/workflows/push-image.yml:38]: I'm not sure if this is the best place for this discussion, but I'm curious what motivation there is for switching to a community organization? Is it just the ability to log in as a member?

**Comment by martyspiewak** [comment]
I think it would be good to resolve [this RFC](https://github.com/paketo-buildpacks/builder/pull/42) before merging this PR, if possible.

File: .github/workflows/push-image.yml (hunk 1 above)
Comment (floragj) [reply .github/workflows/push-image.yml:38]: Yeah @ekcasey I hear you on that. I was sort of on the fence. I'll pop it in a secret for consistency sake. Wrt the transition to a community org, would that mean that we'd just end up having a new username and pass to use for this operation?

**Review by ekcasey — APPROVED** [review]

File: .github/workflows/push-image.yml (hunk 1 above)
Comment (ekcasey) [reply .github/workflows/push-image.yml]: You could still log in as the org itself, but the org can have members and you could also log in as a member. That way you could configure the permissions of both real and bot team members as appropriate. https://docs.docker.com/docker-hub/orgs/

No need to do it now, just thought I would bring mention it as a possibility for the future.

**Review by ekcasey — APPROVED** [review]

**Pull request merged**